lib/db/index.js

```diff
@@ -108,6 +109,32 @@ function preClients() {
   }
 }
 
+function serviceClients() {
+  var clients = config.get('serviceClients');
+  if (clients && clients.length) {
+    logger.debug('serviceClients.loading', clients);
+
+    return P.all(clients.map(function(client) {
+      return exports.getClient(client.id).then(function(existing) {
+        if (existing) {
+          logger.verbose('seviceClients.existing', client);
+          return;
+        }
+
+        return exports.registerClient({
+          id: client.id,
+          name: client.name,
+          hashedSecret: encrypt.hash(unique.secret()),
```

Comment:
No, that pattern was actually a poor idea, and I clarified in bugs that
those should NOT be the hashes for those clients (cause something can
indeed hash to 0), but that the clients should have random hashes that were
thrown away afterwards.

However, this inserts into the db at startup, which @jrgm may object too.

On Wed, Sep 30, 2015, 10:04 PM Ryan Kelly <notifications@github.com> wrote:

> In lib/db/index.js
> <https://github.com/mozilla/fxa-oauth-server/pull/336#discussion_r40879361>
> :
>
> > +function serviceClients() {
> > +  var clients = config.get('serviceClients');
> > +  if (clients && clients.length) {
> > +    logger.debug('serviceClients.loading', clients);
> > +
> > +    return P.all(clients.map(function(client) {
> > +      return exports.getClient(client.id).then(function(existing) {
> > +        if (existing) {
> > +          logger.verbose('seviceClients.existing', client);
> > +          return;
> > +        }
> > +
> > +        return exports.registerClient({
> > +          id: client.id,
> > +          name: client.name,
> > +          hashedSecret: encrypt.hash(unique.secret()),
>
> In other places we've used
> "0000000000000000000000000000000000000000000000000000000000000000" as the
> hashedSecret for clients that aren't supposed to use a secret. Is that a
> pattern we should repeat for these service clients for consistency?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/fxa-oauth-server/pull/336/files#r40879361>.
>